src/workerd/api/sql-test.js

```diff
@@ -322,6 +322,41 @@ async function test(sql) {
   assert.equal(sql.voluntarySizeLimit, 1073741823 * 4096);
   sql.voluntarySizeLimit = 65536;
   assert.equal(sql.voluntarySizeLimit, 65536);
+
+  // Test savepoints, success
+  {
+    await scheduler.wait(1);
+    const { success, result, error } = sql.savepoint(() => {
+      sql.exec("CREATE TABLE IF NOT EXISTS should_succeed (VALUE text);");
+      return "some data"
+    })
```

Comment:
May I suggest a callback-based alternative following the Node.js error-back pattern? For instance,

```
sql.savepoint(() => {
  sql.exec("CREATE TABLE IF NOT EXISTS should_succeed (VALUE text);");
  return "some data"
}, (err, result) => {
  if (err) { /* it failed */ }
  console.log(result);
})
```

This would be far more idiomatic and would work seamlessly with APIs like Node.js `util.promisify` to get a promisified version if someone wanted... e.g. 

```
import { promisify } from 'node:util';

const savepoint = promisify(sql.savepoint.bind(sql));
const result = await savepoint(() => {
  sql.exec("CREATE TABLE IF NOT EXISTS should_succeed (VALUE text);");
  return "some data"
});
```

Reply:
Yeah I agree it should be more standard, but since this is synchronous, I'm going to go with returning/throwing the result of the callback directly.